pepper-apis/src/test/java/org/broadinstitute/ddp/route/GetStudyStatisticsRouteTest.java

```diff
@@ -0,0 +1,251 @@
+import org.broadinstitute.ddp.db.dao.JdbiMailingList;
+import org.broadinstitute.ddp.db.dao.JdbiUserStudyEnrollment;
+import org.broadinstitute.ddp.db.dao.KitTypeDao;
+import org.broadinstitute.ddp.db.dao.StatisticsConfigurationDao;
+import org.broadinstitute.ddp.db.dao.UserGovernanceDao;
+import org.broadinstitute.ddp.db.dto.ActivityInstanceDto;
+import org.broadinstitute.ddp.json.errors.ApiError;
+import org.broadinstitute.ddp.json.statistics.StatisticsFigure;
+import org.broadinstitute.ddp.model.activity.definition.FormActivityDef;
+import org.broadinstitute.ddp.model.activity.definition.FormSectionDef;
+import org.broadinstitute.ddp.model.activity.definition.i18n.Translation;
+import org.broadinstitute.ddp.model.activity.definition.question.PicklistOptionDef;
+import org.broadinstitute.ddp.model.activity.definition.question.PicklistQuestionDef;
+import org.broadinstitute.ddp.model.activity.definition.template.Template;
+import org.broadinstitute.ddp.model.activity.instance.answer.PicklistAnswer;
+import org.broadinstitute.ddp.model.activity.instance.answer.SelectedPicklistOption;
+import org.broadinstitute.ddp.model.activity.revision.RevisionMetadata;
+import org.broadinstitute.ddp.model.activity.types.PicklistRenderMode;
+import org.broadinstitute.ddp.model.activity.types.TemplateType;
+import org.broadinstitute.ddp.model.address.MailAddress;
+import org.broadinstitute.ddp.model.dsm.KitType;
+import org.broadinstitute.ddp.model.statistics.StatisticsType;
+import org.broadinstitute.ddp.model.user.EnrollmentStatusType;
+import org.broadinstitute.ddp.service.AddressService;
+import org.broadinstitute.ddp.util.Auth0Util;
+import org.broadinstitute.ddp.util.ConfigManager;
+import org.broadinstitute.ddp.util.TestDataSetupUtil;
+import org.broadinstitute.ddp.util.TestUtil;
+import org.jdbi.v3.core.Handle;
+import org.junit.AfterClass;
+import org.junit.BeforeClass;
+import org.junit.Test;
+
+public class GetStudyStatisticsRouteTest extends IntegrationTestSuite.TestCase {
+
+    private static TestDataSetupUtil.GeneratedTestData testData;
+    private static Auth0Util.TestingUser testUser2;
+    private static Auth0Util.TestingUser testUser3;
+    private static ActivityInstanceDto instance1Dto;
+    private static ActivityInstanceDto instance2Dto;
+    private static ActivityInstanceDto instance3Dto;
+    private static String user1Guid;
+    private static String user2Guid;
+    private static String user3Guid;
+    private static String token;
+    private static String url;
+    private static Gson gson;
+    private static MailAddress mailAddress;
+    private static long kidRequestId;
+
+    @BeforeClass
+    public static void setup() throws Exception {
+        gson = new Gson();
+        TransactionWrapper.useTxn(handle -> {
+            testData = TestDataSetupUtil.generateBasicUserTestData(handle);
+            testUser2 = TestDataSetupUtil.generateTestUser(handle, testData.getStudyGuid());
```

Comment:
Looks like `generateTestUser()` will create a new user in Auth0. Usually, we try to avoid calling Auth0 because it slows down the tests. And Auth0 has API rate limits so this can occasionally fail and cause the whole test to fail.

Can you try just creating a user directly? E.g. using `UserDao` and related utilities.

Reply:
I see. Fixed